packages/core/src/render3/util/discovery_utils.ts

```diff
@@ -203,6 +218,61 @@ export function getDirectives(element: Element): {}[] {
   return context.directives === null ? [] : [...context.directives];
 }
 
+/**
+ * Metadata for a directive developers can use for debugging purposes or tooling.
+ */
+export interface DirectiveMetadata {
+  inputs: Record<string, string>;
+  outputs: Record<string, string>;
+}
+
+/**
+ * Metadata for a component developers can use for debugging purposes or tooling.
+ */
+export interface ComponentMetadata extends DirectiveMetadata {
+  encapsulation: ViewEncapsulation;
+  onPush: boolean;
+}
```

Comment:
```suggestion
/**
 * Partial metadata for a given component instance.
 * This information might be useful for debugging purposes or tooling.
 * Currently the following fields are available:
 *  - inputs
 *  - outputs
 *  - encapsulation
 *  - changeDetection
 */
export interface ComponentDebugMetadata extends DirectiveDebugMetadata {
  encapsulation: ViewEncapsulation;
  changeDetection: ChangeDetectionStrategy;
}
```
Note: I'd propose align these fields closer to how we have them in the decorator, so I changed `onPush` to `changeDetection`